chaste_codegen/tests/test_linearity_check.py

```diff
@@ -89,77 +85,62 @@ def test_wrong_params_subst_deriv_eqs7(y_derivatives, membrane_voltage_var, non_
-                        "(_second_inward_calcium_current_f_Ca_gate$f_Ca, _environment$time), -_second_inward_calcium_c"
-                        "urrent_f_Ca_gate$alpha_f_Ca*_second_inward_calcium_current_f_Ca_gate$f_Ca + _second_inward_ca"
-                        "lcium_current_f_Ca_gate$beta_f_Ca*(120.0*_intracellular_calcium_concentration$Ca_i*(1.0 - _se"
-                        "cond_inward_calcium_current_f_Ca_gate$f_Ca)/(_intracellular_calcium_concentration$Ca_i + 0.00"
-                        "1) + (1.0 - _second_inward_calcium_current_f_Ca_gate$f_Ca)*(-_intracellular_calcium_concentra"
-                        "tion$Ca_i/(_intracellular_calcium_concentration$Ca_i + 0.001) + 1.0))), Eq(Derivative(_extrac"
-                        "ellular_calcium_concentration$Ca_o, _environment$time), _extracellular_calcium_concentration$"
-                        "K_diff*(-_extracellular_calcium_concentration$Ca_o + _extracellular_calcium_concentration$Cab"
-                        ") - 0.5*(_Na_Ca_exchanger$k_NaCa*(-_extracellular_sodium_concentration$Na_o**_Na_Ca_exchanger"
-                        "$n_NaCa*_intracellular_calcium_concentration$Ca_i*exp_(_membrane$V*(_Na_Ca_exchanger$gamma - "
-                        "1.0)*(_Na_Ca_exchanger$n_NaCa - 2.0)/_membrane$RTONF) + _intracellular_sodium_concentration$N"
-                        "a_i**_Na_Ca_exchanger$n_NaCa*_extracellular_calcium_concentration$Ca_o*exp_(_Na_Ca_exchanger$"
-                        "gamma*_membrane$V*(_Na_Ca_exchanger$n_NaCa - 2.0)/_membrane$RTONF))/((144.92753623188406*_int"
-                        "racellular_calcium_concentration$Ca_i + 1.0)*(_Na_Ca_exchanger$d_NaCa*(_extracellular_sodium_"
-                        "concentration$Na_o**_Na_Ca_exchanger$n_NaCa*_intracellular_calcium_concentration$Ca_i + _intr"
-                        "acellular_sodium_concentration$Na_i**_Na_Ca_exchanger$n_NaCa*_extracellular_calcium_concentra"
-                        "tion$Ca_o) + 1.0)) + _calcium_background_current$g_b_Ca*(-0.5*_membrane$RTONF*log(_extracellu"
-                        "lar_calcium_concentration$Ca_o/_intracellular_calcium_concentration$Ca_i) + _membrane$V) + 4."
-                        "0*_second_inward_calcium_current$P_si*_second_inward_calcium_current_d_gate$d*(1.0 - _second_"
-                        "inward_calcium_current_f_Ca_gate$f_Ca)*(_membrane$V - 50.0)*(-_extracellular_calcium_concentr"
-                        "ation$Ca_o*exp_(-2.0*(_membrane$V - 50.0)/_membrane$RTONF) + _intracellular_calcium_concentra"
-                        "tion$Ca_i*exp_(100.0/_membrane$RTONF))*(-_intracellular_calcium_concentration$Ca_i/(_intracel"
-                        "lular_calcium_concentration$Ca_i + 0.001) + 1.0)/(_membrane$RTONF*(1.0 - exp_(-2.0*(_membrane"
-                        "$V - 50.0)/_membrane$RTONF))))/(_intracellular_calcium_concentration$Ve*_membrane$F))]")
+                                          s_model.get_equations_for)
+    assert str(deq) == \
```

Comment:
No easy way to make this more useful for debugging? Move the string into a file? Compare line by line?

Reply:
Moving the strings to a file is a fair point